src/git-source-provider.ts

```diff
@@ -146,79 +149,110 @@ export async function cleanup(repositoryPath: string): Promise<void> {
   await removeGitConfig(git, authConfigKey)
 }
 
-async function tryPrepareExistingDirectory(
+async function getGitCommandManager(
```

Comment:
i moved some existing logic into this function so the caller's function reads easier.

this function just instantiates the git command manager (wrapper over git executable)